src/tactic/transport.lean

```diff
@@ -0,0 +1,31 @@
+import meta.expr meta.rb_map
+
+namespace tactic
+
+private meta def transport_with_prefix_fun_aux (f : name → option name) (pre tgt_pre : name)
+  (attrs : list name) :
+  name → command :=
+λ src,
+do
+  let tgt := src.map_prefix (λ n, if n = pre then some tgt_pre else none),
+  (get_decl tgt >> skip) <|>
+  do
+    decl ← get_decl src,
+    (decl.type.list_names_with_prefix pre).mfold () (λ n _, transport_with_prefix_fun_aux n),
```

Comment:
don't fold over unit, just use `list.mmap'`

Reply:
This is a `name_set`, not a `list`. There is no `name_set.mmap'`, and `name_set.mmap` only works on functions `f : name → name`.

Reply:
Oh, ok! `mathlib` could define it really easily though :)